pkg/controller/installation/installation_controller.go

```diff
@@ -324,6 +344,36 @@ func (r *ReconcileInstallation) preflightChecks(installation *integreatlyv1alpha
 	logrus.Info("Running preflight checks..")
 
 	result := reconcile.Result{}
+	eventRecorder := r.mgr.GetEventRecorderFor("Preflight Checks")
+
+	if installation.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManaged) {
+		requiredSecrets := []string{installation.Spec.SMTPSecret}
```

Comment:
@aidenkeating @philbrookes This is now the only secret being checked in pre-flight, since @jjaferson's recent change handles the github-oauth-secret in a different way. I'm not sure if this should now be handled in that other way, or if it's still best here.

Reply:
@grdryn I don't believe any changes to this approach have been confirmed. @philbrookes Are you still ok with this approach?

For now the flow would be:
- RHMI installation halts
- SRE complete SOP to create SMTP Secret
- RHMI installation continues

Eventually, the flow would be:
- RHMI installation halts
- Automated SMTP Service creates Secret on cluster
- RHMI installation continues

Reply:
There's maybe another concern here also: the `SMTPSecret` field is optional (because we already do releases of this operator project here on GitHub, so we shouldn't break compatibility & I don't really want to have to support a v1alpha1 and v1alpha2 at the same time (although at v1alpha* I'm not sure if we'd be expected to? Maybe I should just change to v1alpha2 (dropping v1alpha1 immediately) & make it mandatory?).

Reply:
Going to leave it as optional for now. Once we're on 4.3, we can look at adding a default value for it.